Arion/sources/GilbertJohnsonKeerthi.cpp

```diff
@@ -253,7 +253,25 @@ bool intersection::gjk::DoSimplex(gjk::Simplex& simplex, glm::vec3& direction)
     //Check if a current simplex contains the origin
     if (SimplexContainsOrigin(simplex))
     {
-        return true;
+        //FixMe: Note this check prevents (not always) degenerate polytops from going to QHull, but degrates perfomance, is it worth it?
```

Comment:
do you have some numbers/test cases to analyze this?

Reply:
@dsodintsov Not yet, but I do have somewhat related task for it, I think I should have a proper one